src/cache/surface_albedo.jl

```diff
@@ -0,0 +1,195 @@
+"""
+Base.@kwdef struct ConstantAlbedo{FT} <: SurfaceAlbedoModel
+    α::FT = 0.38
+end
+
+"""
+    Base.@kwdef struct RegressionFunctionAlbedo{FT} <: SurfaceAlbedoModel
+
+A regression function-based surface albedo model of Jin et al. (2011, referred to as J11).
+
+
+"""
+Base.@kwdef struct RegressionFunctionAlbedo{FT} <: SurfaceAlbedoModel
+    n::FT = 1.34                                                  # relative refractive index of water and air (n = n_w/n_a) TODO: f(wavelength) for a spectrally dependent scheme
+    n0::FT = 1.34                                                 # refractive index of water for visible light
+    p::Vector{FT} = [
+        0.0152,
+        -1.7873,
+        6.8972,
+        -8.5778,
+        4.071,
+        -7.6446,
+        0.1643,
+        -7.8409,
+        -3.5639,
+        -2.3588,
+        10.0538,
+    ]                                                             # regression coefficients of J11 for eq. 4
+    wave_slope::Function = u -> sqrt(FT(0.003) + FT(0.00512) * u) # mean wave slope distribution width (Cox-Munk model, J11 Eq. 2)
+    q_clear::Vector{FT} = [-0.1482, -0.012, 0.1608, -0.02244]     # regression coefficients of J11 for eq. 5a
+    q_cloud::Vector{FT} = [-0.1479, -0.1502, -0.016]              # regression coefficients of J11 for eq. 5b
+end
+
+"""
+    set_surface_albedo!(Y, p, t, α_model::ConstantAlbedo)
+
+Set the surface albedo to a constant value.
+"""
+function set_surface_albedo!(Y, p, t, α_model::ConstantAlbedo{FT}) where {FT}
+
+    (; direct_sw_surface_albedo, diffuse_sw_surface_albedo) =
+        p.radiation.radiation_model
+
+    @. direct_sw_surface_albedo = α_model.α
+    @. diffuse_sw_surface_albedo = α_model.α
+end
+
+"""
+    set_surface_albedo!(Y, p, t, α_model::RegressionFunctionAlbedo{FT})
+
+Wrapper to call the formulations of Jin et al. (2011) for direct and diffuse surface albedos of the ocean.
+"""
+function set_surface_albedo!(
+    Y,
+    p,
+    t,
+    α_model::RegressionFunctionAlbedo{FT},
+) where {FT}
+
+    (; direct_sw_surface_albedo, diffuse_sw_surface_albedo, cos_zenith) =
+        p.radiation.radiation_model
+
+    λ = FT(0) # spectral wavelength (not used for now)
+    μ = cos_zenith
+
+    u_int = Fields.field_values(Fields.level(Y.c.uₕ, 1))
+    int_local_geometry_values =
+        Fields.field_values(Fields.level(Fields.local_geometry_field(Y.c), 1))
+
+    wind_speed = p.scratch.ᶠtemp_field_level
+    wind_speed_values = Fields.field_values(wind_speed)
+    @. wind_speed_values =
+        Geometry._norm.(
+            CT12.(u_int, int_local_geometry_values),
+            int_local_geometry_values,
+        )
+    # TODO: Is there a neater way of defining wind speed? E.g., do we want to use SF.jl windspeed? including gustiness?  Or do we want to extrapolate to the surface using the MOST profile? It would be good to expose SF.surface_conditions in that case.
```

Comment:
Currently the windspeed is an internal function in the SF package; There is additionally a `recover_profile` method if you wish to use a known L_MO to compute windspeed in the surface layer.

Reply:
Sounds good. I opened an issue for this: https://github.com/CliMA/ClimaAtmos.jl/issues/2693